doc/classes/ProjectSettings.xml

```diff
@@ -461,11 +461,13 @@
 		<member name="debug/gdscript/warnings/confusable_local_usage" type="int" setter="" getter="" default="1">
 			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when an identifier that will be shadowed below in the block is used.
 		</member>
-		<member name="debug/gdscript/warnings/constant_used_as_function" type="int" setter="" getter="" default="1">
+		<member name="debug/gdscript/warnings/constant_used_as_function" type="int" setter="" getter="" default="1" is_deprecated="true">
 			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a constant is used as a function.
+			[i]Deprecated.[/i] This warning is never produced. Instead, an error is generated if the expression type is known at compile time.
```

Comment:
I don't know how to feel about this. If they literally do not work it is better to remove them outright, otherwise it can get very confusing. There's a lot of "unordered" settings in the warning section and these contribute to clutter.

There's no compatibility broken here by removing them. Users that overrode the default setting for whatever reason will still keep this setting in the `.project` file, albeit they would no longer be built-in.
If the reason to deprecate them is to tell the user that these settings are not built-in, there should probably be a better way of going about it that is outside the scope of this PR...

Reply:
> There's no compatibility broken here by removing them.

Hypothetically, someone could use `@warning_ignore()`, although that wouldn't make sense. Removing the warnings will result in script error:

![](https://github.com/godotengine/godot/assets/47700418/d847bf63-335f-451e-bc91-63f1a39f1980)

Reply:
I see now. So it is hardcoded this way. A shame it has to be this way but it's better to document these as _deprecated_ than a compilation error I suppose, even though the ignore never did anything in practice.